### ISSUE TITLE
serverless trigger to create the runners like those in github hosted runners

### ISSUE THREAD

**Issue by nnWhisperer** [terraform-google-modules/terraform-google-github-actions-runners#200]
### TL;DR

Hello,
As far as I understood, solutions hosted in this repo are either scalable with a kubernetes cluster or to be manually run to create runner instances when needed. GKE unfortunately needs a VM instance to be run to manage itself. My proposal/feature request is to add a serverless (webhook/http based) structure to trigger this creation instead. Thereby, users wouldn't need to manage runners themselves and the system would be auto-scaled when necessary. Thank you

### Detailed design

```markdown
Possibly adding webhooks from github to a serverless gcp function that trigger creation of runner instances.
```

### Additional information

I think this solution would make using gcp for self-hosted runners more viable and competitive compared to the current status and would cause more traction from github hosted runner users.
